frontend/views/containers/contributions/SupportHistory.vue

```diff
@@ -15,31 +15,65 @@ div(:class='isReady ? "" : "c-ready"')
 </template>
 
 <script>
+import { mapGetters } from 'vuex'
 import { humanDate } from '@model/contracts/shared/time.js'
 import BarGraph from '@components/graphs/BarGraph.vue'
 
 export default ({
   name: 'GroupSupportHistory',
   data () {
     return {
-      isReady: false,
-      history: []
+      isReady: false
     }
   },
   components: {
     BarGraph
   },
-  created () {
-    // Todo replace history with real data
-    const testNumber = 6
-    for (let i = testNumber; i > 0; i--) {
-      const date = new Date()
-      date.setMonth(date.getMonth() - i)
-      this.history.push({
-        total: 1 / testNumber * (testNumber - i + 1),
-        title: humanDate(date, { month: 'long' })
+  computed: {
+    ...mapGetters([
+      'groupSettings',
+      'currentPaymentPeriod',
+      'periodBeforePeriod',
+      'paymentsForPeriod'
+    ]),
+    mincome () {
+      return this.groupSettings.mincomeAmount
+    },
+    periods () {
+      const periods = []
+      let lastPeriod = this.currentPaymentPeriod
+      for (let i = 0; i < 5; i++) {
+        periods.unshift(lastPeriod)
+        lastPeriod = this.periodBeforePeriod(lastPeriod)
+      }
+      periods.unshift(lastPeriod)
+      return periods
+    },
+    history () {
+      return this.periods.map((period, i) => {
+        const payments = this.paymentsForPeriod(period)
+        const { totalDistributionAmount, numberOfNeedyPeople } = this.parsePayments(payments)
+        return {
+          total: numberOfNeedyPeople === 0 ? 1 : totalDistributionAmount * this.mincome / numberOfNeedyPeople,
```

Comment:
In English, it sounds rather politically incorrect (and somewhat funny) to name this variable `numberOfNeedyPeople`. It is also a bit against the ethos of the project's core values - which is to recognize that everyone in the group has something of value to offer, even if it isn't money.

So can we rename this variable to `numReceivers` instead?

Same change requested below inside of `parsePayments`.

Reply:
Sure. I agree with you. It's really better for me too.